src/core/net/srp_server.cpp

```diff
@@ -323,9 +326,11 @@ bool Server::HasNameConflictsWith(Host &aHost) const
 
         for (const Host &host : mHosts)
         {
-            if (host.HasServiceInstance(service.GetInstanceName()))
+            if (host.HasServiceInstance(service.GetInstanceName()) &&
```

Comment:
@abtink I think you change here to match against instance name. But shouldn't the full name be used here? For example, `abcdef._meshcop._udp.local` should not conflict with `abcdef._ipps._tcp.local`.

The SRP draft 12 statement about this is:
> [2.3.3. ](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-12.html#section-2.3.3)[FCFS Name And Signature Validation](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-12.html#name-fcfs-name-and-signature-val)
Assuming that a DNS Update message has been validated with these conditions and is a valid SRP Update, the server checks that the name in the Host Description Instruction exists. If so, then the server checks to see if the KEY record on that name is the same as the KEY record in the Host Description Instruction. The server performs the same check for the KEY records in any Service Description Instructions.

I think name in the service description instruction is the full name but not the instance name. Thoughts?

Reply:
I think it should be the full name.

It should be fine to have both `Floor 1._printer._tcp` and `Floor 1._light._tcp`.

Reply:
Yes. I agree. It should be full name (let's do it from a different PR).

Reply:
Digging a bit more, I realized we may be already doing the check correctly in the code 😃 

`Service::GetInstanceName()` returns the full service instance name and not just the "instance label". So we are checking for full name (e.g. `abc._serv1._udp` in  `host.HasServiceInstance(service.GetInstanceName()`).

```c++

        /**
         * This method gets the full service instance name of the service.
         *
         * @returns  A pointer service instance name (as a null-terminated C string).
         *
         */
        const char *GetInstanceName(void) const { return mDescription->mInstanceName.AsCString(); }
```

My suggestion is we try to harmonize our use of the terms "name" and "label" in the code (I think we may be using it in most places) 
- Basically "name" always refers to full name (not a single label)
- And use the term "label" when we refer to just one label (e.g. service instance label, or `abc` in `abc._serv1._udp`).

Reply:
- Added PR https://github.com/openthread/openthread/pull/7588 to add this test (two clients registering services with same instance label).

Reply:
I also did not realize it for some time. I actually wrote a quick test-case, expecting it to fail, and was surprised why it was not failing  😄

Reply:
> I also did not realize it for some time. I actually wrote a quick test-case, expecting it to fail, and was surprised why it was not failing 😄

I guess now we have a real issue to fix.